File: 7_July/23_Jul/leaderboard.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Player
{
    public string Name { get; set; }
    public int Score { get; set; }
}

class Leaderboard
{
    private List<Player> players = new List<Player>();

    public void AddPlayer(string name, int score)
    {
        players.Add(new Player { Name = name, Score = score });
        players = players.OrderByDescending(p => p.Score).ToList();
    }

    public void DisplayLeaderboard()
    {
        Console.WriteLine("Leaderboard:");
        for (int i = 0; i < players.Count; i++)
        {
            Console.WriteLine($"{i + 1}. {players[i].Name} - {players[i].Score} points");
        }
    }
}

class Program
{
    static void Main(string[] args)
    {
        Leaderboard leaderboard = new Leaderboard();

        leaderboard.AddPlayer("Alice", 100);
        leaderboard.AddPlayer("Bob", 85);
        leaderboard.AddPlayer("Charlie", 95);
        leaderboard.AddPlayer("David", 110);

        leaderboard.DisplayLeaderboard();
    }
}

File: 7_July/22_Jul/leaderboard.cc
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>

struct Player {
    std::string name;
    int score;
};

bool compareScores(const Player& a, const Player& b) {
    return a.score > b.score;
}

class Leaderboard {
private:
    std::vector<Player> players;

public:
    void addPlayer(const std::string& name, int score) {
        players.push_back({name, score});
        std::sort(players.begin(), players.end(), compareScores);
    }

    void displayLeaderboard() {
        std::cout << "Leaderboard:\n";
        for (size_t i = 0; i < players.size(); ++i) {
            std::cout << i + 1 << ". " << players[i].name << " - " << players[i].score << " points\n";
        }
    }
};

int main() {
    Leaderboard leaderboard;

    leaderboard.addPlayer("Alice", 100);
    leaderboard.addPlayer("Bob", 85);
    leaderboard.addPlayer("Charlie", 95);
    leaderboard.addPlayer("David", 110);

    leaderboard.displayLeaderboard();

    return 0;
}